modules/govuk/manifests/user.pp

```diff
@@ -68,9 +74,12 @@
     $membership = 'minimum'
   }
 
+  validate_re($expiry, '^(absent|\d{4}-\d{2}-\d{2})$')
```

Comment:
Not a biggie, but it's a shame Puppet doesn't seem to have a function to validate a date specifically. I am not sure what the behaviour would be if an invalid date is inputted (like 2016-02-30).